### PR TITLE
Add new blog post related to XCP-NG and template

### PR DESCRIPTION
**Pull request by mickaelbaron**
Hi,

Please find in this PR, a contribution to add a new blog post related to XCP-NG and how to create and use custom XCP-NG templates.

Best

> Before submitting the pull request, you must agree with the following statements by checking both boxes with a 'x'.
> * [x] "I accept that my contribution is placed under the CC BY-SA 2.0 license [1]."
> * [x] "My contribution complies with the Developer Certificate of Origin [2]." 
>
> [1] https://creativecommons.org/licenses/by-sa/2.0/
> [2] https://docs.xcp-ng.org/project/contributing/#developer-certificate-of-origin-dco

### CODE REVIEW

**Comment by olivierlambert** [comment]
Thanks @mickaelbaron 

Putting this to review :+1:

**Comment by olivierlambert** [comment]
Also @Danp2 can help to review :)

**Review by Fohdeesha — COMMENTED** [review]
due to family issues I have no time to review documentation currently

**Comment by olivierlambert** [comment]
No worries, Dan will do it :+1:

**Review by Danp2 — CHANGES_REQUESTED** [review]

File: docs/guides/create-use-custom-xcpng-ubuntu-templates.md
Comment (mickaelbaron) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md:3]: I followed the questions (description) used in other guides. https://docs.xcp-ng.org/category/guides/

I propose this update: How to create and use custom XCP-NG templates for Ubuntu?

File: docs/guides/create-use-custom-xcpng-ubuntu-templates.md
Comment (mickaelbaron) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md:16]: I meant that what I demonstrate can also be applied to other distributions like Debian, for example as long as Cloud-init is supported

Comment (Danp2) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md:16]: I was only questioning the repeated linking to external sites for things like Cloud-init. Should all instances be links or only the first one? @olivierlambert Is there a preference here?

File: docs/guides/create-use-custom-xcpng-ubuntu-templates.md
Comment (mickaelbaron) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md:14]: Oki

File: docs/guides/create-use-custom-xcpng-ubuntu-templates.md
Comment (mickaelbaron) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md:23]: Ok

File: docs/guides/create-use-custom-xcpng-ubuntu-templates.md
Comment (mickaelbaron) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md:42]: Ok, thanks

File: docs/guides/create-use-custom-xcpng-ubuntu-templates.md
Comment (mickaelbaron) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md:16]: Sorry, I used to use links for each time I mention a tool (if link is existing).

File: docs/guides/create-use-custom-xcpng-ubuntu-templates.md
Comment (mickaelbaron) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md:42]: I used a "full" impersonal structure

In order to access the ISO file during the virtual machine creation step, it needs to be placed in an ISO storage repository. There are several types available (Local, NFS, or SMB). For this post, the first type will be used. Feel free to browse through this [post](https://xcp-ng.org/blog/2022/05/05/how-to-create-a-local-iso-repository-in-xcp-ng/) to learn how to create a local ISO storage repository. In the following, I assume that there is a local ISO storage repository on the XCP-NG host.

File: docs/guides/create-use-custom-xcpng-ubuntu-templates.md
Comment (mickaelbaron) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md:45]: To be honest, I do not use this solution. I upload the ISO file by scp.

However, I will take your comment into account.

File: docs/guides/create-use-custom-xcpng-ubuntu-templates.md
Comment (mickaelbaron) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md:47]: Ok

File: docs/guides/create-use-custom-xcpng-ubuntu-templates.md
Comment (olivierlambert) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md:16]: I think it's a detail we can ignore

**Review by Danp2 — CHANGES_REQUESTED** [review]

File: docs/guides/create-use-custom-xcpng-ubuntu-templates.md
Comment (mickaelbaron) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md:53]: Oops, big mistake, sorry and thank you. It will be fixed in the next commit.

Comment (olivierlambert) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md:53]: Don't worry, reviews are meant for this.

Comment (MrGrymReaper) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md]: > I've also experimented with Ubuntu 24.04. I suppose it was a bad copy/paste.

When you originally wrote the text you copy and pasted Focal Fossa release code name of Ubuntu 20.04, Jammy Jellyfish is the release code name of 22.04 and the code name for 24.04 is Noble Numbat. FYI

Comment (mickaelbaron) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md]: Damn, I'm lost with these code names, I shouldn't have specified it because it induces errors and nobody cares.

**Review by MrGrymReaper — CHANGES_REQUESTED** [review]
There really needs to specify that the release is an LTS release of Ubuntu being used as part of this guide. This is especially important for if the system requirements change in later releases, as well as package names change in relation to this guide.

The LTS releases of Ubuntu are supported for a longer period of time in relation to the standard support releases. This thus means that the package names tend to be fairly set during that time and don't change often.

Between standard supported releases the package names can change more often. Also the standard supported releases of Ubuntu have the latest versions of software, while the LTS are an older more stable production test long term supported version.

So any commands required may not work between releases or can be different.

**Comment by stormi** [comment]
Should this be a guide inside Xen Orchestra's documentation instead? These templates are not directly an XCP-ng feature.

**Comment by olivierlambert** [comment]
> Should this be a guide inside Xen Orchestra's documentation instead? These templates are not directly an XCP-ng feature.

Sometimes it's hard to find the right place between both. Frankly, at some point, a unified doc would make more sense. But until then, I think that would be OK to get this one into XCP-ng doc.

File: docs/guides/create-use-custom-xcpng-ubuntu-templates.md
Comment (mickaelbaron) [reply docs/guides/create-use-custom-xcpng-ubuntu-templates.md]: Hi,

I updated the documentation to fix this problem.

Mickael

**Comment by mickaelbaron** [comment]
Hello,

What’s the status of my pull request?

**Comment by mickaelbaron** [comment]
Hi,

I would like to know if my PR is still relevant

Best

**Comment by stormi** [comment]
I think so, based on @olivierlambert's comments.

**Comment by olivierlambert** [comment]
Sorry for being very late on this, we were under too much water. Merging it ASAP.

**Review by olivierlambert — APPROVED** [review]

**Comment by olivierlambert** [comment]
@stormi I don't know why there's a review stuck while it was answered and why the DCO isn't correctly checked? :thinking:

**Comment by stormi** [comment]
The PR doesn't look in a good state, from the git point of view. Too many commits in the branch, which should likely be rebased on master, and not contain a merge commit like it does now.

**Comment by stormi** [comment]
I don't understand @Danp2's merge commit :thinking:

**Comment by stormi** [comment]
I'll try to fix it.

**Comment by stormi** [comment]
(Ideally, @Danp2 should review the changes made after his comments and Approve the PR)

**Comment by stormi** [comment]
I don't know why it closed the PR after my force-push to remove the merge commit.

**Comment by stormi** [comment]
Looks like there's something I don't understand in how github works with forks and pushes to PRs. I'll create a new PR with these changes.

**Pull request closed without merging**
